code/modules/assembly/flash.dm

```diff
@@ -190,3 +190,32 @@
 	item_state = "nullrod"
 
 /obj/item/device/assembly/flash/handheld //this is now the regular pocket flashes
+
+/obj/item/device/assembly/flash/armimplant
+	name = "photon projector"
+	desc = "A high-powered photon projector implant normally used for lighting purposes, but also doubles as a flashbulb weapon. Self-repair protocals fix the flashbulb if it ever burns out."
+	var/flashcd = 20
+	var/overheat = 0
+	var/obj/item/organ/cyberimp/arm/flash/I = null
+
+/obj/item/device/assembly/flash/armimplant/burn_out()
+	if(I)
+		I.owner << "<span class='warning'>Your photon projector implant overheats and deactivates!</span>"
+		I.Retract()
+	overheat = 1
+	addtimer(src, "cooldown", flashcd * 2)
+
+/obj/item/device/assembly/flash/armimplant/try_use_flash(mob/user = null)
+	if(overheat)
+		if(I)

```

Comment:
combine these two if checks into one

    if(I && I.owner)

Also, why do you not check I.owner exists in other procs, why just this one?